src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseDataFileServiceTest.cs

```diff
@@ -1684,15 +1892,183 @@ await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContex
+                },
+            };
+
+            var releaseFiles = new List<ReleaseFile>
+            {
+                new ()
+                {
+                    Release = release,
+                    File = new File { Type = FileType.Data },
+                    Order = 0,
+                },
+                new ()
+                {
+                    Release = release,
+                    File = new File { Type = FileType.Data },
+                    Order = 1,
+                },
+                new ()
+                {
+                    Release = release,
+                    File = new File { Type = FileType.Data },
+                    Order = 3,
+
+                },
+                new () // Ancillary files should be ignored
+                {
+                    Release = release,
+                    File = new File { Type = FileType.Ancillary },
+                    Order = 5,
+                },
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            var statisticsDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.Releases.AddRangeAsync(release);
+                await contentDbContext.ReleaseFiles.AddRangeAsync(releaseFiles);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var dataFormFile = CreateFormFileMock(dataFileName).Object;
+            var metaFormFile = CreateFormFileMock(metaFileName).Object;
+            var blobStorageService = new Mock<IBlobStorageService>(MockBehavior.Strict);
+            var fileUploadsValidatorService = new Mock<IFileUploadsValidatorService>(MockBehavior.Strict);
+            var dataImportService = new Mock<IDataImportService>(MockBehavior.Strict);
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbContextId))
+            {
+                fileUploadsValidatorService
+                    .Setup(s => s.ValidateSubjectName(release.Id, subjectName))
+                    .ReturnsAsync(Unit.Instance);
+
+                fileUploadsValidatorService
+                    .Setup(s => s.ValidateDataFilesForUpload(
+                        release.Id,
+                        dataFormFile,
+                        metaFormFile,
+                        null))
+                    .ReturnsAsync(Unit.Instance);
+
+                dataImportService
+                    .Setup(s => s.Import(
+                        It.IsAny<Guid>(),
+                        It.Is<File>(file => file.Type == FileType.Data && file.Filename == dataFileName),
+                        It.Is<File>(file => file.Type == Metadata && file.Filename == metaFileName)))
+                    .ReturnsAsync(new DataImport { Status = QUEUED }); // @MarkFix
```

Comment:
Is ther esomething that you still need to fix up here?

Reply:
That was me during the nightmare merge conflict - and then I forgot about it. I'll sort it.